### PR TITLE
mdspan: new package

### PR DESCRIPTION
**Pull request by Andrew-Dunning-NNL**
Adds new package for [kokkos mdspan](https://github.com/kokkos/mdspan)

### CODE REVIEW

**Comment by Andrew-Dunning-NNL** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/mdspan/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    variant("examples", default=True, description="Enable examples")
+    variant("tests", default=False, description="Enable tests")
+    variant("benchmarks", default=False, description="Enable benchmarks")
+
+    depends_on("benchmark", when="+benchmarks")
+    depends_on("googletest@main", when="+tests")
+
+    version("stable", branch="stable")
+    version("0.6.0", sha256="d6b7b9d4f472106df1d28729bd8383a8a7ea7938adf9f82d3be9c151344830d9")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/mdspan/package.py]:
```suggestion
    version("stable", branch="stable")
    version("0.6.0", sha256="d6b7b9d4f472106df1d28729bd8383a8a7ea7938adf9f82d3be9c151344830d9")
    
    variant("examples", default=True, description="Enable examples")
    variant("tests", default=False, description="Enable tests")
    variant("benchmarks", default=False, description="Enable benchmarks")

    depends_on("benchmark", when="+benchmarks")
    depends_on("googletest@main", when="+tests")
```

Small nitpick, but it makes it easier to review packages if they follow the common format of 
```
1. versions
2. variants
3. dependencies
```

**Review by alecbcs — CHANGES_REQUESTED** [review]
@Andrew-Dunning-NNL thanks for the new package. This looks great! Just one quick recommendation.

File: var/spack/repos/builtin/packages/mdspan/package.py (hunk 1 above)
Comment (Andrew-Dunning-NNL) [reply var/spack/repos/builtin/packages/mdspan/package.py]: oh good call, Thanks!

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks for the quick revisions @Andrew-Dunning-NNL.

**Pull request merged**